mungegithub/mungers/submit-queue.go

```diff
@@ -647,6 +657,34 @@ func (sq *SubmitQueue) getQueueStatus() []byte {
 	return sq.marshal(status)
 }
 
+// getMetrics returns a json representation of important stats about the
+// submit queue's health, for external monitoring/graphing/alerting.
+func (sq *SubmitQueue) getMetrics() []byte {
+	sq.Lock()
+	defer sq.Unlock()
+
+	status := sq.e2e.GetBuildStatus()
+	blocked := false
+	for _, state := range status {
+		if state.Status != "Stable" {
+			blocked = true
+		}
+	}
```

Comment:
When my other PR is merged, there will be caching for GCSBasedStable.

Don't rewrite this logic. Cache it from the place that it's used and just repeat the cached value here. ...or wait. Use promethieus. Don't make your own metrics system. Just kick the metric from where the computation is done.